### PR TITLE
Add Dropdown Transitions

### LINKED ISSUE #874: Add Dropdown Transitions
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.
Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

**To Do**: Add a nice transition effect to the step collapse using bootstrap collapse plugin.

### 📋 Step by Step
- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [defaultHtmlStepUi js](https://github.com/publiclab/image-sequencer/blob/main/examples/lib/defaultHtmlStepUi.js) in the `image-sequencer` repository (press the little pen Icon) and edit the line as shown below. 

Change this code
https://github.com/publiclab/image-sequencer/blob/e1a113cde183a620eaa9bb0fd0c37520178eb84a/examples/lib/defaultHtmlStepUi.js#L158-L161

[See this page](https://publiclab.github.io/community-toolbox/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
    $(step.ui.querySelector(".toggle")).on("click", () => {
      $(step.ui.querySelector('.toggleIcon')).toggleClass('fa-caret-up').toggleClass('fa-caret-down');
- $(step.ui.querySelectorAll(".cal")).toggleClass("collapse");
+ $(step.ui.querySelectorAll(".step")).collapse('toggle');
    });
```
- [ ] 💾 **Commit** your changes.

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review
### Please keep us updated
💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?
🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by mgroovyank**
Fixes #874 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by mgroovyank** [comment]
@HarshKhandeparkar  Please review it.
ImageSequencerGoogleChrome

**Comment by HarshKhandeparkar** [comment]
Did you do this in the browser?

**Comment by mgroovyank** [comment]
Yes.

**Comment by HarshKhandeparkar** [comment]
Can you also change the dist files also? You can commit to the same branch in the browser.

**Comment by HarshKhandeparkar** [comment]
If you commit to the same branch, the changes will be visibme here.

**Comment by mgroovyank** [comment]
But you said not to change dist files?

**Comment by HarshKhandeparkar** [comment]
I told you not to touch change the dist files directly but it seems like you are doing this from the browser and cannot use geunt for building the dist files. Sorry for the trouble you have to take but you will have to to this in the browser or clone locally and build them using grunt. Sorry.

**Comment by HarshKhandeparkar** [comment]
No need to change the minified files. Only the browserified files need to be changed.

**Comment by mgroovyank** [comment]
I will do it in the browser right now.

**Comment by HarshKhandeparkar** [comment]
We are planning on adding the dist files to gitignore soon.

**Comment by mgroovyank** [comment]
Dist files do create a confusion for any beginner.

**Comment by mgroovyank** [comment]
@HarshKhandeparkar  added changes to dist files.

**Comment by Harshithpabbati** [comment]
Ha but the changes are in dist files.

**Comment by HarshKhandeparkar** [comment]
The changes are there in both the dist and source. You looked at the commit which adds dist

**Comment by Harshithpabbati** [comment]
Fine cool evenrything is nice !!!

**Comment by HarshKhandeparkar** [comment]
Do you approve? If yes, please add the label. Thanks

**Comment by Harshithpabbati** [comment]
Wait a minute why is it not working in local.

**Comment by Harshithpabbati** [comment]
see
see this.

**Comment by Harshithpabbati** [comment]
Image is not hiding.

**Comment by HarshKhandeparkar** [comment]
Did you clear the cache?

**Comment by Harshithpabbati** [comment]
Yeah I did it.

File: examples/lib/defaultHtmlStepUi.js [hunk 1]
```diff
@@ -157,7 +157,7 @@ function DefaultHtmlStepUi(_sequencer, options) {
     }
     $(step.ui.querySelector(".toggle")).on("click", () => {
       $(step.ui.querySelector('.toggleIcon')).toggleClass('fa-caret-up').toggleClass('fa-caret-down');
-      $(step.ui.querySelectorAll(".cal")).toggleClass("collapse");
+       $(step.ui.querySelectorAll(".cal")).collapse('toggle');
```
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js]: Change the selector from '.cal' to '.step', that was the problem

File: dist/image-sequencer-ui.js [hunk 2]
```diff
@@ -510,7 +510,7 @@ function DefaultHtmlStepUi(_sequencer, options) {
     }
     $(step.ui.querySelector(".toggle")).on("click", () => {
       $(step.ui.querySelector('.toggleIcon')).toggleClass('fa-caret-up').toggleClass('fa-caret-down');
-      $(step.ui.querySelectorAll(".cal")).toggleClass("collapse");
+      $(step.ui.querySelectorAll(".cal")).collapse('toggle');
```
Comment (HarshKhandeparkar) [inline dist/image-sequencer-ui.js]: Change the selector from '.cal' to '.step', that was the problem

**Comment by Harshithpabbati** [comment]
And run grunt uglify and grunt browserify. And push the dist files.

**Comment by HarshKhandeparkar** [comment]
I think he is changing the files from the browser o he cannot use grunt. So he has to manually change the dist files.

File: dist/image-sequencer-ui.js (hunk 2 above)
Comment (mgroovyank) [reply dist/image-sequencer-ui.js]: > Change the selector from '.cal' to '.step', that was the problem

Doing that will collapse the whole step including the toggle button.

Comment (HarshKhandeparkar) [reply dist/image-sequencer-ui.js]: Why doesn't the image collapse then?

**Comment by Harshithpabbati** [comment]
Then let him send only the files leave about the dist files. I will send the dist files.

**Comment by HarshKhandeparkar** [comment]
How will you commit the fist files?

**Comment by Harshithpabbati** [comment]
Let him make it we will think about it.

**Comment by mgroovyank** [comment]
My mistake. I missed one change.sorry.

**Comment by HarshKhandeparkar** [comment]
Ok.

**Comment by HarshKhandeparkar** [comment]
@Harshithpabbati please check the changes locally. Also if @mgroovyank can update the dist files(since it is a small change). Please do so

**Comment by HarshKhandeparkar** [comment]
Forget the minified files for now. They are of no use to use right now. They will be gitignored soon. Only the browserified files are needed so that the changes are visible.

**Comment by Harshithpabbati** [comment]
Soory for that I can't do it now. I am going back to my hostel. I will be online from tommorow.
Thanks

**Comment by HarshKhandeparkar** [comment]
Ok np.

**Comment by HarshKhandeparkar** [comment]
@mgroovyank can you please share a new GIF. Sorry for the trouble.

**Comment by mgroovyank** [comment]
@HarshKhandeparkar I will share it soon.

**Comment by HarshKhandeparkar** [comment]
Can someone test this out locally? Maybe @Divy123?

**Comment by mgroovyank** [comment]
I am not sharing the GIF as the last commits don't change anything. It is working perfectly fine for me.

**Comment by HarshKhandeparkar** [comment]
We would still like to have a final GIF on `localhost:3000` since the last one you shared is on sequencer.publiclab.org and I don't even know how you managed to change that.

**Comment by Divy123** [comment]
Surely I am testing it out now.

**Comment by HarshKhandeparkar** [comment]
Thank you!

**Comment by Divy123** [comment]
ezgif com-video-to-gif


This is something I am getting now.

**Comment by Harshithpabbati** [comment]
Yeah thats the problem. I am asking for that.

**Comment by Harshithpabbati** [comment]
And I have a doubt how do you take the video ???
I mean the gif??

**Comment by Divy123** [comment]
I cloned his fork.
And then used simple screen recorder app to have this.

**Comment by Harshithpabbati** [comment]
Thanks 😅

**Comment by mgroovyank** [comment]
I don't know if this will help but my commits on my forked repository are on a branch named- **patch-3**

**Comment by HarshKhandeparkar** [comment]
I think the image ia not contained inside the div that is collapsed. Try giving the img tag a class of `.cal` `.collapse` and `.in`

**Comment by HarshKhandeparkar** [comment]
Or maybe the div is removed from the dom but the image exists?? Someone will have to check the dev tools

**Comment by Divy123** [comment]
I think the image should be made to display none in case when we want to collapse the step.
I looked into the dev tools and found the image was still there even after collapsing the step.

**Comment by mgroovyank** [comment]
In my console it shows this.

Screenshot (13)

**Comment by HarshKhandeparkar** [comment]
How are you using sequencer.publiclab.org?

**Comment by mgroovyank** [comment]
@HarshKhandeparkar  I am doing the changes in _elements_ tab and in the _event listeners_ associated with every element on the webpage.

Screenshot (14)

**Comment by HarshKhandeparkar** [comment]
Please don't do this. Try out your changes locally as this site sequencer.publiclab.org is atleast 20d old as compared to main branch of the repo.

**Comment by Harshithpabbati** [comment]
Or else you can change in my gh-pages the link to it is harshithpabbati.github.io/image-sequencer

**Comment by HarshKhandeparkar** [comment]
Oh you can initialize your github pages as well

**Comment by mgroovyank** [comment]
@HarshKhandeparkar  check this out. I have added the _cal_ and _collapse in_ class to the anchor tag which displays the image. And it works.

ImageSequencerGoogleChrome

**Comment by Harshithpabbati** [comment]
Ok this time it looks good.

**Comment by Harshithpabbati** [comment]
@HarshKhandeparkar now it's working fine :)

**Comment by HarshKhandeparkar** [comment]
Did you check it out locally? If you are sure that it works then add the label. Thanks @mgroovyank

**Comment by Harshithpabbati** [comment]
Yeah I checked it locally.

**Comment by Harshithpabbati** [comment]
Thanks @mgroovyank

**Comment by Harshithpabbati** [comment]
We need to wait for this to merge https://github.com/publiclab/image-sequencer/pull/867. And then we should have this I guess.

**Comment by HarshKhandeparkar** [comment]
@jywarren this is ready to be merged. A first timer pr.

**Comment by Harshithpabbati** [comment]
Let this merge first.

**Comment by mgroovyank** [comment]
@HarshKhandeparkar @Harshithpabbati Thanks for your help.

**Comment by Harshithpabbati** [comment]
@HarshKhandeparkar but the main thing was we were not asking for dist files.

**Comment by HarshKhandeparkar** [comment]
@jywarren are you merging this?

**Comment by jywarren** [comment]
Yep, thank you!!!

**Pull request merged**